server/routes/prisonerChangeIncentiveLevelDetails.ts

```diff
@@ -0,0 +1,144 @@
+    const { prisonerNumber } = req.params
+    const systemToken = await hmppsAuthClient.getSystemClientToken(res.locals.user.username)
+    const prisonApi = new PrisonApi(systemToken)
+    const incentivesApi = new IncentivesApi(systemToken)
+    const { errors, formValues = {} } = pageData || {}
+
+    try {
+      const prisonerDetails = await prisonApi.getPrisonerDetails(prisonerNumber)
+      const { agencyId, bookingId, firstName, lastName } = prisonerDetails
+      const incentiveLevelDetails: IncentiveSummaryForBookingWithDetails =
+        await incentivesApi.getIncentiveSummaryForPrisoner(prisonerNumber)
+      const currentIncentiveLevel: string = incentiveLevelDetails.iepLevel
+      const prisonIncentiveLevels = await incentivesApi.getPrisonIncentiveLevels(agencyId)
+      const selectableLevels = prisonIncentiveLevels.map(level => ({
+        text: currentIncentiveLevel === level.levelName ? `${level.levelName} (current level)` : level.levelName,
+        value: level.levelCode,
+        checked: level.levelCode === formValues.newIepLevel,
+      }))
+
+      return res.render('pages/prisonerChangeIncentiveLevelDetails.njk', {
+        agencyId,
+        bookingId,
+        breadcrumbPrisonerName: putLastNameFirst(firstName, lastName),
+        currentIncentiveLevel,
+        errors,
+        formValues,
+        prisonerNumber,
+        prisonerName: formatName(firstName, lastName),
+        profileUrl: `${res.app.locals.dpsUrl}/prisoner/${prisonerNumber}`,
+        selectableLevels,
+      })
+    } catch (error) {
+      res.locals.redirectUrl = `incentive-level-details/prisoner/${prisonerNumber}`
+      throw error
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-expect-error
+  const renderConfirmation = async (req, res) => {
+    const { prisonerNumber } = req.params
+    const systemToken = await hmppsAuthClient.getSystemClientToken(res.locals.user.username)
+    const prisonApi = new PrisonApi(systemToken)
+    const incentivesApi = new IncentivesApi(systemToken)
+
+    try {
+      const prisonerDetails = await prisonApi.getFullDetails(prisonerNumber, true)
+      const { agencyId, bookingId, firstName, lastName, assignedLivingUnit } = prisonerDetails
+      const locationId: string | undefined = assignedLivingUnit?.description
+      const incentiveSummary = await incentivesApi.getIncentiveSummaryForPrisoner(prisonerNumber)
+      const nextReviewDate =
+        incentiveSummary?.nextReviewDate && moment(incentiveSummary.nextReviewDate, 'YYYY-MM-DD HH:mm')
+
+      return res.render('pages/prisonerChangeIncentiveLevelConfirmation.njk', {
+        agencyId,
+        bookingId,
+        breadcrumbPrisonerName: putLastNameFirst(firstName, lastName),
+        cancelUrl: '/',
+        incentiveSummary,
+        manageIncentivesUrl:
+          agencyId && locationId && locationId.includes('-')
+            ? `/incentive-summary/${agencyId}-${locationId.split('-')[0]}`
+            : '/',
```

Comment:
Yes good point` / ` just takes you back to incentives homepage which isn't where you would have started the journey. Not sure about `/select-location`. I mean its only a fall back but would it make more sense to return to the prisoners profile page?

Reply:
but this link says “Return to incentive reviews”, it's the _other_ one that says “Return to prisoner profile page”! this one should get to as near as it can to the reviews table – in this case it cannot get to it directly so /select-location is the next best thing in my opinion